### PR TITLE
Enforce constructor visibility

### LINKED ISSUE #979: Constructor ignores visibility modifier
For any of the visibilities (internal, external, private, public):

```
contract Test {
    function Test() internal {
    }
}
```

the following ABI is generated: `[{"inputs":[],"type":"constructor"}]`, and contract deployment works without hiccups.

I think it would make sense rejecting any visibility modifier on the constructor. Are there any cases currently where a non-public constructor is useful? We do not have static methods, so I don't see a way a non-public constructor could be executed.

### PR DESCRIPTION
**Pull request by axic**
Fixes #979

### CODE REVIEW

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 1]
```diff
@@ -3872,6 +3872,27 @@ BOOST_AUTO_TEST_CASE(unsatisfied_version)
...
 }
 
+BOOST_AUTO_TEST_CASE(constant_constructor)
+{
+	char const* text = R"(
+		contract test {
+			function test() constant {}
+		}
+	)";
+	BOOST_CHECK(expectError(text, true) == Error::Type::SyntaxError);
```
Comment (chriseth) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp]: Should be `TypeError`

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 2]
```diff
@@ -3872,6 +3872,27 @@ BOOST_AUTO_TEST_CASE(unsatisfied_version)
...
+}
+
+BOOST_AUTO_TEST_CASE(external_constructor)
+{
+	char const* text = R"(
+		contract test {
+			function test() external {}
+		}
+	)";
+	BOOST_CHECK(expectError(text, true) == Error::Type::SyntaxError);
```
Comment (chriseth) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp]: Should be `TypeError`.

**Comment by chfast** [comment]
LGTM

**Comment by axic** [comment]
Actually I wanted #1018 to be merged first, and this rebased on top of it. (Because this contains the changes of #1018 too)

**Comment by axic** [comment]
Oh we need to include this in the changelog.

**Pull request merged**